File: python/euler/bin/p0055.py
# project euler: problem 55


def compute() -> str:
    cnt = 0
    for n in range(10_000):
        i = n
        for _ in range(50):
            i += int(str(i)[::-1])
            s = str(i)
            if s == s[::-1]:
                break
        else:
            cnt += 1

    return str(cnt)


def solve() -> str:
    return compute()

File: typescript/euler/bin/p0055.ts
// project euler: problem 55

import { isPalindrome, range } from "../lib/util.ts";

export const compute = (limit: number): string => {
  let acc = 0;

  loop_i:
  for (let i of range(0, limit)) {
    for (const _ of range(0, 50)) {
      i += Number(String(i).split("").reverse().join(""));
      if (isPalindrome(i) === true) {
        continue loop_i;
      }
    }
    acc += 1;
  }

  return String(acc);
};

export const solve = (): string => compute(10_000);
